simulator/src/main/java/com/hedera/block/simulator/BlockStreamSimulatorApp.java

```diff
@@ -30,14 +36,14 @@ public class BlockStreamSimulatorApp {
     private static final System.Logger LOGGER =
             System.getLogger(BlockStreamSimulatorApp.class.getName());
 
-    Configuration configuration;
-    BlockStreamManager blockStreamManager;
-    PublishStreamGrpcClient publishStreamGrpcClient;
-    BlockStreamConfig blockStreamConfig;
+    private final Configuration configuration;
+    private BlockStreamManager blockStreamManager;
+    private PublishStreamGrpcClient publishStreamGrpcClient;
+    private BlockStreamConfig blockStreamConfig;
```

Comment:
IMO, these should become final. They are only instantiated in constructors, which have the same logic. We could use constructor overloading, which is always a good idea. Also, the final keyword not only makes our code more strict, but actually improves performance by making our field immutable and there are hints to the compiler/JVM which work like magic to make our programs faster. Small things do compound and make a difference.